crates/holochain/tests/publish/mod.rs

```diff
@@ -0,0 +1,64 @@
+use holo_hash::ActionHash;
+use holochain::core::workflow::publish_dht_ops_workflow::num_still_needing_publish;
+use holochain::sweettest::{
+    consistency_60s, SweetConductorBatch, SweetConductorConfig, SweetDnaFile,
+};
+use holochain_wasm_test_utils::TestWasm;
+use std::time::Duration;
+
+/// Verifies that publishing terminates naturally when enough validation receipts are received.
+#[cfg(feature = "test_utils")]
+#[tokio::test(flavor = "multi_thread")]
+async fn publish_termination() {
+    let _g = holochain_trace::test_run().unwrap();
+    const NUM_CONDUCTORS: usize = 6; // Need 5 peers to send validation receipts back
```

Comment:
I think this is another problem. We shouldn't be mandating a minimum network size (not least because this will struggle on CI 😟). I think the publish workflow should actually be smarter and check the set of validation receipts against the current network peers then only republish if there's a new peer who might send back a validation receipt. It makes for more work locally but saves network and we could actually avoid broadcasting to peers we've already asked to send us validation receipts.

This is less critical than the bug being fixed here but still worth a look at some point I think.

Reply:
A lot of our design decisions were made assuming large networks... I'm not sure this is currently possible with the way publish delegation works. We'd need to include the list of nodes we already have receipts for in the publish itself so the nodes that are delegating don't in turn send to those transitive peers.

Reply:
Ahh interesting okay, yeah that would complicate things. Maybe just something to bear in mind about test scenarios etc then and we could return to this if it becomes a problem rather than a possibility for optimization :)